### PR TITLE
MKL using static linking

### PR DESCRIPTION
**Pull request by massich**
This PR squashes #253 and modifies the building calls just to make travis green. There are no modifications to the building system.

### CODE REVIEW

File: build_tools/travis/script.sh [hunk 1]
```diff
@@ -20,6 +20,10 @@ if [[ "$ENABLE_PACKAGING" == "1" ]]; then
     # Remove conda used from wrapping
     rm -rf ${HOME}/miniconda_wrap
 
+    if [[ "$BLASLAPACK_IMPLEMENTATION" == "MKL" ]]; then
+        sudo rm -rf /opt/intel
+    fi
```
Comment (agramfort) [inline build_tools/travis/script.sh]: in this script replace:

/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/uninstall)"

by:

```
curl -O https://raw.githubusercontent.com/Homebrew/install/master/uninstall
chmod +x 
uninstall
./uninstall --force
```

that's why travis fails on mac

File: .travis.yml [hunk 2]
```diff
@@ -36,19 +47,32 @@ matrix:
...
     - os: linux
       compiler: gcc
       env:
         - USE_SYSTEM=1
         - USE_PROJECT=1
-        - USE_VTK=1
+        - USE_VTK=0
         - USE_OMP=1
         - USE_GIFTI=1
         - USE_COVERAGE=1
```
Comment (agramfort) [inline .travis.yml]: I would not put coverage on for this build. I would activate it just for the most complete build. wdyt?

**Comment by agramfort** [comment]
LGTM. Let's see if CIs turn green !

File: .travis.yml (hunk 2 above)
Comment (massich) [reply .travis.yml]: doesn't it merge them ?

**Comment by agramfort** [comment]
Maybe...

**Comment by agramfort** [comment]
I have restarted mac build again but it fails as it takes more than 10 mins to download the MKL and our build script does not print anything during the 10 mins... can we have a verbose download?

**Comment by papadop** [comment]
Yes. It's a parameter or file(DOWNLOAD ). I think It's PROGRESS_BAR. I removed it but it is easy to add it backLe 11 déc. 2017 2:26 AM, Alexandre Gramfort <notifications@github.com> a écrit :I have restarted mac build again but it fails as it takes more than 10 mins to download the MKL and our build script does not print anything during the 10 mins... can we have a verbose download?

**Comment by massich** [comment]
@agramfort travis seem do die [here](https://travis-ci.org/openmeeg/openmeeg/jobs/314387009#L3761). This is long after installing MKL

File: .travis.yml [hunk 3]
```diff
@@ -72,3 +96,5 @@ env:
     - secure: XBNTHU+DVCKeUKjxVSfvAkFr/KnvzYDDn28u8Y41v9kT+yHE1HHPnSYJS9JndQllOUAXtT0Ix81LSKG9MVthIWs5uPTOOg4hPZloE8mOYWSs/EHLraEEbZFUAaoCz2ZVYdA92HM1YAVo4Bi+zMy9g7Ra4POSWwFY5n/zV+ma//A=
     - secure: NwGYiI9vDCLsIN+uAJRPlUV9g4RraaHfNiJRGBmsc8kOaPvBL2fb3Mia6nOfkfcaVOmkEPhWeAOW8XKHqVj6982SBQrbnS/xfU/EHiDJaq7YVBRfkCn4/9yswyhCxvrS97JagIgnEjlhbIV9VLZ19A/e2fhWEzGbJabLyZ0ZNBc=
     - secure: ZXoAGZAYFC/kPwiX+TEpNX+rh2zVrVIw6BDdd/Qt1SiBB3+BZWwosqh+Wv1mLpjce7IAOXO42rg/OUwfPMl2egjFHth8OU827dyfYC0yFsgfN6myWD+HK+U6/3i+fp5WknWOhXzzFc72CgBwYNCnrIKNJ5LUMESUyew+vIgo5p0=
+    - MAKEFLAGS="-j 2"
+    - USE_PROGRESSBAR=ON
```
Comment (agramfort) [inline .travis.yml]: I don't think it's this progressbar.

File: cmake/InstallMKL.cmake [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    set(MKL_BASE_NAME "m_mkl_2017.4.220")
+    set(MKL_INSTALLER_ARCHIVE "${MKL_BASE_NAME}.dmg")
+else()
+    set(MKL_URL_DIR 12070)
+    set(MKL_BASE_NAME "l_mkl_2018.0.128")
+    set(MKL_INSTALLER_ARCHIVE "${MKL_BASE_NAME}.tgz")
+endif()
+
+set(MKL_BASE_URL "http://registrationcenter-download.intel.com/akdlm/irc_nas/tec")
+file(DOWNLOAD "${MKL_BASE_URL}/${MKL_URL_DIR}/${MKL_INSTALLER_ARCHIVE}" ${CMAKE_BINARY_DIR}/${MKL_INSTALLER_ARCHIVE}
```
Comment (agramfort) [inline cmake/InstallMKL.cmake]: it's here. Use SHOW_PROGRESS option cf:

https://cmake.org/cmake/help/v3.0/command/file.html

**Comment by agramfort** [comment]
travis is green !!!

let me see now if I can make appveyor green again by reverting changes there.

**Comment by agramfort** [comment]
ok it's green on travis (with MKL) and appveyor using (openblas)

I suggest to merge and take care of windows MKL in a next PR

@massich ok for you?

**Comment by massich** [comment]
@agramfort is fine with me.

**Pull request merged**